### PR TITLE
Fix Windows issue. Empty feature added.

### PR DESCRIPTION
**Pull request by jonasho**
Fixes Issue #15 with windows, where backslash is used instead of slash.

Also added new feature to delete all files inside the bucket before uploading (#18).

### CODE REVIEW

**Comment by funkybob** [comment]
I really appreciate you taking on these two features! Not having Windows, I couldn't work on that one.

All of the formatting changes [and pointless stripping of ;] has made this PR very very noisy, and hides the actual changes.

I'll look more into it later today before merging.

Can you confirm you've tested this code in a non-trivial use? {I would normally assume that, but someone else didn't and I got bitten by that}

**Comment by jonasho** [comment]
Thanks for getting back so quick. I've updated the PR to only reflect the real changes not the noisy formatting changes. Sorry for that.

I've tested it with windows and Linux both are working fine. (With nested files and directories)

The empty bucket code was tested with a few hundred files also inside Subdirectories and in the root folder. 
I hope I've covered everything but obviously no guarantee ;)

File: index.js [hunk 1]
```diff
@@ -110,43 +139,61 @@ class Assets {
...
+                const body = fs.readFileSync(path.join(opt.source, filename))
+                const type = mime.lookup(filename) || opt.defaultContentType || 'application/octet-stream'
+
+                this.log(`\tFile:  ${filename} (${type})`);
+
+                // when using windows path join resolves to backslashes, but s3 is expecting a slash
+                // therefore replace all backslashes with slashes
+                const key = path
+                  .join(prefix, filename)
+                  .split('\\')
```
Comment (funkybob) [inline index.js]: Why not just a replace, instead of split+join? 

`
.replace(/\\/g, '/')
`

Comment (jonasho) [reply index.js]: No particular reason. I've updated the PR to use `replace`

**Pull request merged**